mon-pix/app/components/authentication/oidc-provider-selector.gjs

```diff
@@ -10,10 +10,12 @@ export default class OidcProviderSelector extends Component {
   get providerOptions() {
     const { providers = [] } = this.args;
 
-    const options = providers.map((provider) => ({
-      label: provider.organizationName,
-      value: provider.id,
-    }));
+    const options = providers
+      .filter((provider) => provider.isVisible)
+      .map((provider) => ({
+        label: provider.organizationName,
+        value: provider.id,
+      }));
```

Comment:
Plutôt que parcourir 2 fois le tableau, utiliser une fonction qui parcoure le tableau 1 seule fois :
```suggestion
    const options = providers.reduce(
      (acc, provider) =>
        provider.isVisible ? [...acc, { label: provider.organizationName, value: provider.id }] : acc,
      [],
);
```

Reply:
Personnellement, le reduce, me donne assez mal au crane 😅

Reply:
Les reduce sont toujours plus difficile à lire mais une fois testé on est assuré de leur bon fonctionnement. On peut aussi utiliser `.flatMap` pour obtenir un parcours de tableau avec 1 seule itération cf. https://2ality.com/2022/05/processing-arrays-non-destructively.html , mais je trouve que c'est un tout petit tordre l'intention de `.flatMap`. Alors que `.reduce` est clairement fait pour l'utilisation dont il est question ici.

Reply:
ça me va, peut être peut on mettre le reduce dans une sous fonction au nom évocateur ?